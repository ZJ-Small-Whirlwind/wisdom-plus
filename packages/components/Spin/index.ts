import { withInstall } from '@wisdom-plus/utils/with-install'
import Spin from './src/spin'

export const WpSpin = withInstall(Spin)
export default WpSpin

export * from './src/spin'