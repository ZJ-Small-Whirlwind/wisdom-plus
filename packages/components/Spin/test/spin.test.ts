import Spin from '../src/spin'
import { mount } from '@vue/test-utils'

describe('Attrs', () => {
    test('Add class', () => {
        const wrapper = mount(Spin, {
            props: {
                class: "test"
            }
        })

        expect(wrapper.find('.wp-spin').classes()).toContain('test')
    })
})