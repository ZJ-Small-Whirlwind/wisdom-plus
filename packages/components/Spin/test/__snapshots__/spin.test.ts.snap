// Vitest Snapshot v1

exports[`Attrs > Add class > class="test" 1`] = `"<teleport-stub class=\\"test\\"></teleport-stub>"`;

exports[`Attrs > Add class > class="test" 2`] = `"<teleport-stub class=\\"test\\"></teleport-stub>"`;

exports[`Attrs > Add class > test 1`] = `"<teleport-stub class=\\"test\\"></teleport-stub>"`;

exports[`Attrs > Add class 1`] = `"<teleport-stub></teleport-stub>"`;

exports[`Attrs > Add class 2`] = `"<teleport-stub></teleport-stub>"`;
